src/libraries/System.Globalization/tests/Invariant/InvariantMode.cs

```diff
@@ -1216,6 +1216,12 @@ public void TestCasing()
 
             for (int i = 0; i <= 0xFFFF; i++)
             {
+                // Surrogates are characters in the Unicode range U+D800—U+DFFF
+                if (i >= 0xD800 && i <= 0xDFFF)
+                {
+                    continue;
```

Comment:
IMO we shouldn't be skipping these tests via a _continue_ statement. It's perfectly legal to call ToUpper / ToLower on an unpaired surrogate char, and the behavior that we want is that we leave the char alone. If we want to skip the surrogate range in this test, then we should have another test specifically for the surrogate range, regression testing that we are in fact reflecting the input chars as-is.